src/test/regress/spec/isolation_metadata_sync_vs_all.spec

```diff
@@ -0,0 +1,205 @@
+}
+
+step "s1-start-metadata-sync"
+{
+	SELECT start_metadata_sync_to_node('localhost', 57638);
+}
+
+session "s2"
+
+step "s2-begin"
+{
+	BEGIN;
+}
+
+step "s2-commit"
+{
+	COMMIT;
+}
+
+step "s2-alter-table"
+{
+	ALTER TABLE dist_table ADD COLUMN z int;
+}
+
+step "s2-add-fk"
+{
+	ALTER TABLE dist_table ADD CONSTRAINT y_fk FOREIGN KEY (y) REFERENCES ref_table(y);
+}
+
+step "s2-drop-fk"
+{
+	ALTER TABLE dist_table DROP CONSTRAINT y_fk;
+}
+
+step "s2-drop-table"
+{
+	DROP TABLE dist_table;
+}
+
+step "s2-create-dist-table"
+{
+	CREATE TABLE new_dist_table(id int, data int);
+	SELECT create_distributed_table('new_dist_table', 'id');
+}
+
+step "s2-create-ref-table"
+{
+	CREATE TABLE new_ref_table(id int, data int);
+	SELECT create_reference_table('new_ref_table');
+}
+
+step "s2-attach-partition"
+{
+	ALTER TABLE dist_partitioned_table ATTACH PARTITION dist_partitioned_table_p1 FOR VALUES FROM (1) TO (9);
+}
+
+step "s2-detach-partition"
+{
+	ALTER TABLE dist_partitioned_table DETACH PARTITION dist_partitioned_table_p1;
+}
+
+step "s2-create-partition-of"
+{
+	CREATE TABLE dist_partitioned_table_p2 PARTITION OF dist_partitioned_table FOR VALUES FROM (10) TO (20);
+}
+
+step "s2-create-type"
+{
+	CREATE TYPE my_type AS (a int, b int);
+}
+
+step "s2-drop-type"
+{
+	DROP TYPE my_type;
+}
+
+step "s2-create-dist-func"
+{
+	CREATE FUNCTION squares(int) RETURNS SETOF RECORD
+    AS $$ SELECT i, i * i FROM generate_series(1, $1) i $$
+    LANGUAGE SQL;
+
+	SELECT create_distributed_function('squares(int)');
+}
+
+step "s2-drop-dist-func"
+{
+	DROP FUNCTION squares(int);
+}
+
+session "s3"
+
+step "s3-compare-snapshot"
+{
+	SELECT count(*) = 0 AS same_metadata_in_workers
+	FROM
+	(
+		(
+			SELECT unnest(activate_node_snapshot())
+				EXCEPT
+			SELECT unnest(result::text[]) AS unnested_result
+			FROM run_command_on_workers($$SELECT activate_node_snapshot()$$)
+		)
+	UNION
+		(
+			SELECT unnest(result::text[]) AS unnested_result
+			FROM run_command_on_workers($$SELECT activate_node_snapshot()$$)
+				EXCEPT
+			SELECT unnest(activate_node_snapshot())
+		)
+	) AS foo;
+}
+
+step "s3-debug"
+{
+	SELECT unnest(activate_node_snapshot());
+
+	SELECT unnest(result::text[])
+	FROM run_command_on_workers('SELECT activate_node_snapshot()');
+}
+
+// the following operations get blocked when a concurrent metadata sync is in progress
+permutation "s1-begin" "s2-begin" "s1-start-metadata-sync" "s2-alter-table" "s1-commit" "s2-commit" "s3-compare-snapshot"
+permutation "s1-begin" "s2-begin" "s1-start-metadata-sync" "s2-drop-table" "s1-commit" "s2-commit" "s3-compare-snapshot"
+permutation "s1-begin" "s2-begin" "s1-start-metadata-sync" "s2-create-dist-table" "s1-commit" "s2-commit" "s3-compare-snapshot"
+permutation "s1-begin" "s2-begin" "s1-start-metadata-sync" "s2-create-ref-table" "s1-commit" "s2-commit" "s3-compare-snapshot"
+permutation "s1-begin" "s2-begin" "s1-start-metadata-sync" "s2-attach-partition" "s1-commit" "s2-commit" "s3-compare-snapshot"
```

Comment:
should we add "s1-start-metadata-sync" vs "s2-start-metadata-sync" as well? We might have such test in another file. If so, we can skip addinng that here

Reply:
I created another section above for these tests.